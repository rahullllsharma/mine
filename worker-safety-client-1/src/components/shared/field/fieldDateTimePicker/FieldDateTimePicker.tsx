import type { ForwardedRef } from "react";
import type { FieldDateTimePickerProps } from "../BaseFieldDateTimePicker";
import { forwardRef } from "react";
import BaseFieldDateTimePicker from "../BaseFieldDateTimePicker";

export type FieldDatePickerProps = Omit<
  FieldDateTimePickerProps,
  "type" | "icon"
> &
  Required<Pick<FieldDateTimePickerProps, "name">>;

/**
 * The FieldDateTimePicker built for the worker-safety-client.
 *
 * @description
 * For this project, we can use a simpler component that reuses the native
 * _input[type="datetime-local"]_ for everything (agreeded with the design team).
 *
 *
 * **HOWEVER** This component is *NOT* silica ready!
 * For that, we need to build a custom picker that will be consistent across desktop devices while
 * still provinding a good experience for mobile users.
 *
 * @see [MDN input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time)
 */
function FieldDateTimePicker(
  props: FieldDatePickerProps,
  ref: ForwardedRef<HTMLInputElement>
): JSX.Element {
  return (
    <BaseFieldDateTimePicker
      {...props}
      ref={ref}
      type="datetime-local"
      // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#handling_browser_support
      pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}"
    />
  );
}

export default forwardRef(FieldDateTimePicker);
