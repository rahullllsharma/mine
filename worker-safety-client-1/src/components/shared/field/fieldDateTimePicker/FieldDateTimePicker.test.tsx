import { render, screen } from "@testing-library/react";
import FieldDateTimePicker from "./FieldDateTimePicker";

describe(FieldDateTimePicker.name, () => {
  it('should render with the "FieldDateTimePicker" component', () => {
    const { asFragment } = render(
      <FieldDateTimePicker name="datetime-picker" label="Lorem ipsum" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the default value passed", () => {
    render(
      <FieldDateTimePicker
        name="datetime-picker"
        label="Lorem ipsum"
        defaultValue="2022-01-11T10:20"
      />
    );

    screen.getByDisplayValue("2022-01-11T10:20");
  });

  it("should render the value passed", () => {
    render(
      <FieldDateTimePicker
        name="datetime-picker"
        label="Lorem ipsum"
        value="2022-01-11T10:20"
      />
    );

    screen.getByDisplayValue("2022-01-11T10:20");
  });

  it("should be accessable by a label", () => {
    render(
      <FieldDateTimePicker
        name="datetime-picker"
        label="Lorem ipsum"
        value="2022-01-11T10:20"
      />
    );

    expect(screen.getByLabelText(/lorem ipsum/i)).toHaveDisplayValue(
      "2022-01-11T10:20"
    );
    expect(screen.getByLabelText(/lorem ipsum/i)).toHaveValue(
      "2022-01-11T10:20"
    );
  });
});
