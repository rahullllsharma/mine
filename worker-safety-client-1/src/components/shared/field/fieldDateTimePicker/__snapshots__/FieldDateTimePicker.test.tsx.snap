// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined should render with the "FieldDateTimePicker" component 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
      for="datetime-picker"
    >
      Lorem ipsum
    </label>
    <div
      class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid text-neutral-shade-58 border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
    >
      <i
        aria-hidden="true"
        class="ci-calendar ml-2 pointer-events-none w-6 h-6 text-xl leading-none bg-white text-neutral-shade-58"
      />
      <input
        aria-errormessage=""
        aria-invalid="false"
        class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 root p-2"
        id="datetime-picker"
        name="datetime-picker"
        pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}"
        type="datetime-local"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
